src/liboslexec/llvm_gen.cpp

```diff
@@ -3019,6 +3019,69 @@ LLVMGEN(llvm_gen_getattribute)
                                                : rop.llvm_load_string(ustring());
     llvm::Value* attr_name_arg = rop.llvm_load_value(Attribute);
 
+    ustring object_name    = (object_lookup && ObjectName.is_constant())
+                                 ? ObjectName.get_string()
+                                 : ustring();
+    ustring attribute_name = Attribute.is_constant() ? Attribute.get_string()
+                                                     : ustring();
+
```

Comment:
I think if the attribute name isn't constant, since a constant getattribute("", result); can be pretty trivially turned into a "just fail" whereas if it was not constant, we want to do the run-time lookup.  I bet it's pretty easy to get into a place where a "default" shader ends up doing a getattribute("",result) that can basically be used to skip a bunch of work, but if there's an attribute set in the shading node to cause that to turn into getattribute("mycoolval", result), there could be a lot of other stuff that would get run.  If a non-constant attribute name causes us to call build_attribute_getter with "", we can no longer optimize the cases where it's a constant "".

Reply:
The other thing we could do is just pass in the Symbol& and let the renderer inspect them to find that info out.

Reply:
> The other thing we could do is just pass in the Symbol& and let the renderer inspect them to find that info out.

If the name isn't constant, what could the renderer figure out from the Symbol&?  (not rhetorical)

Reply:
Sorry, my first comment was a bit nonsensical.  It could discriminate between a non-const name and a "".

Reply:
After a bit of discussion with Larry and Alex, our proposal here would be to handle the compile time constant empty attribute name string internally, it would always produce a fail. In such case the `build_attribute_getter` function would not be called. The reason being is that a constant empty attribute name is not very sensical. With this in place, you could then be sure that on the renderer side if you receive an empty attribute name, the only possibility is because it's not known at compile time.

I'd prefer to avoid passing LLVM symbols out into user facing code if possible.